parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/RewriteOptions.java

```diff
@@ -39,11 +36,20 @@
 
 /**
  * A set of options to create a ParquetRewriter.
+ *
+ * TODO find a place where to put a proper description of functionality as it is not trivial:
+ * ParquetRewriter allows to stitch files with the same schema into a single file.
+ * Note that ParquetRewriter also can be used for effectively stitch/joining multiple parquet files with
+ * different schemas.
+ * You can provide the main input file group and multiple right side ones. That is possible when:
+ * 1) the number of rows in the main and extra input groups are the same,
+ * 2) the ordering of rows in the main and extra input groups is the same.
  */
 public class RewriteOptions {
 
   private final ParquetConfiguration conf;
   private final List<InputFile> inputFiles;
+  private final List<List<InputFile>> inputFilesR;
```

Comment:
Could we consolidate the internal member variables `inputFiles` and `inputFilesR`. My rough idea is to model these files into `HorizontalInputFile` and `VerticalInputFile`.
- VerticalInputFile: An `InputFile` of a single parquet file which may have partial or full schema.
- HorizontalInputFile: one or more `VerticalInputFile`s with same number rows but different schemas to form a logical wide file. It can choose one `VerticalInputFile` to be the left side and the others to be the right side in your join algorithm. If it contains only one `VerticalInputFile`, this is what we have today.

WDYT?

Reply:
So the RewriteOptions will contain a holder for List<VerticalInputFile> & List<HorizontalInputFile> or some sort of general List<StackableInputFile> that both VerticalInputFile & HorizontalInputFile need to implement? 


By the way. In my current implementation only the total number of rows of the left and on the right need to match. Individual files don't need the row count matched. It seem you suggest putting the same number of records into each VerticalInputFiles group. I'm not sure that particular requirement is easy to satisfy in practical environment. For example in Spark there are ways to preserve the ordering as I showed in the linked Gist in PR description but controlling the number of rows in each output file in Spark is not trivial as far as I know.

Reply:
That make sense. Files on the right side do not need to make number of rows to the left side.